### PR TITLE
Prince and Tai shopping list

### PR DESCRIPTION
**Pull request by Magmanat**
Made index.tsx and ShoppingAnimegang.styled.ts to make the base look of the shopping list

image

### CODE REVIEW

**Comment by Magmanat** [comment]
So far this is the progress for our shopping list without using redux.

Added a card for our group:
image


Full shopping list view:
image


User can delete items from the list:
image


User can add quantity by pressing plus button:
image


User can reduce quantity by pressing minus button:
image


User can add new item by pressing + Add Item button which will bring in an input window:
image

image

image

File: src/pages/ShoppingAnimegang/index.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  TopRow,
+  TotalPricePart,
+} from './styles/ShoppingAnimegang.styled'
+import { Table, Button, Space, Input } from 'antd'
+import { FullScreenContainer } from './styles/ShoppingAnimegang.styled'
+import { PlusOutlined, MinusOutlined } from '@ant-design/icons'
+
+// const handleIncreaseItemCount = () => {
+// }
+// onClick={() => handleIncreaseItemCount()}
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/index.tsx]: remove unused comments

File: src/pages/ShoppingAnimegang/index.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        item.quantity += 1
+      }
+      return item
+    })
+    setShoppingList(newList)
+  }
+  const handleOnMinusClick = (record: Item) => {
+    const newList: Item[] = shoppingList.filter((item) => {
+      if (item.key === record.key) {
+        item.quantity = item.quantity == 0 ? 0 : item.quantity - 1
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/index.tsx]: use `===` instead of `==`

File: src/pages/ShoppingAnimegang/index.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+      }
+      return item
+    })
+    setShoppingList(newList)
+  }
+  const handleOnDeleteClick = (record: Item) => {
+    const newList: Item[] = shoppingList.filter((item) => item.key != record.key)
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/index.tsx]: use `!==` instead

File: src/pages/ShoppingAnimegang/index.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    setShoppingList(newList)
+  }
+  const handleOnAddItemClick = () => {
+    setShowInput(true)
+  }
+  const handleOnType = (e: string) => {
+    setAddItemName(e)
+    console.log(addItemName)
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/index.tsx]: remove console logs

File: src/pages/ShoppingAnimegang/index.tsx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  }
+  const handleOnEnter = () => {
+    if (addItemName != '') {
+      const newList: Item[] = shoppingList.filter((item) => item)
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/index.tsx]: what does this line do? seems like it returns everything in the shoppingList without filtering

File: src/pages/ShoppingAnimegang/index.tsx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    let number = 0
+    for (let i = 0; i < mylist.length; i += 1) {
+      number += mylist[i].total
+    }
+    setTotalShoppingListPrice(number)
+    return number
+  }
+  const onInputChange = (record: Item) => (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newList: Item[] = shoppingList.filter((item) => {
+      if (item.key == record.key) {
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/index.tsx]: use 3 =

File: src/pages/ShoppingAnimegang/styles/ShoppingAnimegang.styled.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  align-items: center;
+  justify-content: start;
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  left: 0;
+  top: 0;
+  padding-top: 50px;
+  padding-left: 50px;
+  padding-right: 50px;
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/styles/ShoppingAnimegang.styled.ts]: consider using padding: 50px 50px auto 50px

File: src/pages/ShoppingAnimegang/index.tsx (hunk 5 above)
Comment (kktai1512) [reply src/pages/ShoppingAnimegang/index.tsx]: yeah, i just want a copy of the list

File: src/pages/ShoppingAnimegang/index.tsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
+import React, { useState } from 'react'
+import {
+  AddItemsContainer,
+  ShoppingListHeader,
+  TablePart,
+  TopRow,
+  TotalPricePart,
+} from './styles/ShoppingAnimegang.styled'
+import { Table, Button, Space, Input } from 'antd'
+import { FullScreenContainer } from './styles/ShoppingAnimegang.styled'
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/index.tsx]: import this w the rest in line 2

File: src/pages/ShoppingAnimegang/index.tsx (hunk 2 above)
Comment (hui444) [reply src/pages/ShoppingAnimegang/index.tsx]: this is still using 2 =, pls use 3 instead

File: src/pages/ShoppingAnimegang/index.tsx (hunk 5 above)
Comment (hui444) [reply src/pages/ShoppingAnimegang/index.tsx]: consider using `..newList: Item[] = {...shoppingList}` instead

File: src/pages/ShoppingAnimegang/index.tsx [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+    <FullScreenContainer>
+      <TopRow>
+        <ShoppingListHeader> Shopping list</ShoppingListHeader>
+        <AddItemsContainer>
+          {!showInput && (
+            <Button onClick={() => handleOnAddItemClick()} type="primary">
+              + Add item
+            </Button>
+          )}
+          {showInput && (
```
Comment (hui444) [inline src/pages/ShoppingAnimegang/index.tsx]: can consider using `? : ` instead - ie `showInput ? <true result> : <false result>`

**Review by hui444 — COMMENTED** [review]
rmb to switch the == into ===

File: src/pages/ShoppingListAnimeGang/index.tsx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+        item.quantity += 1
+      }
+      return item
+    })
+    setShoppingList(newList)
+  }
+  const handleOnMinusClick = (record: Item) => {
+    const newList: Item[] = shoppingList.filter((item) => {
+      if (item.key === record.key) {
+        item.quantity = item.quantity == 0 ? 0 : item.quantity - 1
```
Comment (hui444) [inline src/pages/ShoppingListAnimeGang/index.tsx]: use 3 = instead

**Review by hui444 — APPROVED** [review]
fix the == to === and so resolve conflicts before merging

**Pull request closed without merging**